chainer/functions/math/maximum.py

```diff
@@ -0,0 +1,59 @@
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer import utils
+from chainer.utils import type_check
+
+
+class Maximum(function.Function):
+    """Element-wise maximum of input variables."""
+
+    def check_type_forward(self, in_types):
+        type_check.expect(
+            in_types.size() == 2,
+            in_types[0].dtype.kind == 'f',
+            in_types[1].dtype.kind == 'f',
```

Comment:
I think it is safer to impose `in_types[0].dtype == in_types[1].dtype` because we can ensure the dtype of the output.